atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraServiceTest.java

```diff
@@ -245,6 +245,26 @@ public void selectsFromAllHostsIfDatacenterMappingNotAvailable() {
         assertThat(suggestedHosts).containsExactlyInAnyOrderElementsOf(allHosts);
     }
 
+    @Test
+    public void getRandomHostByActiveConnections() {
+        Set<CassandraServer> servers = generateServers(24);
+        try (CassandraService service = clientPoolWithParams(servers, servers, 1.0)) {
+            service.setLocalHosts(servers.stream().limit(8).collect(ImmutableSet.toImmutableSet()));
+            Set<CassandraServer> desired = servers.stream().limit(3).collect(Collectors.toSet());
+            for (int i = 0; i < 1_000_000; i++) {
+                assertThat(service.getRandomHostByActiveConnections(desired)).isPresent();
```

Comment:
do we want to be a bit stricter and do something like:
```
                Optional<CassandraServer> server =
                        service.getRandomHostByActiveConnections(desired);
                assertThat(server).isPresent();
                assertThat(desired).contains(server.get());
```
?

Reply:
updated and tweaked assert